client/azure_client_set.go

```diff
@@ -4,19 +4,18 @@ import (
 	"errors"
 	"net/http"
 
-	"github.com/giantswarm/microerror"
-	"github.com/giantswarm/micrologger"
-
 	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-04-01/compute"
 	"github.com/Azure/azure-sdk-for-go/services/dns/mgmt/2017-09-01/dns"
 	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2017-09-01/network"
 	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-02-01/resources"
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/Azure/go-autorest/autorest/adal"
 	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
 )
 
-type AzureConfig struct {
+type AzureClientSetConfig struct {
```

Comment:
It is called `AzureClientSet` and `NewAzureClientSet`.

Reply:
Nitpick: I think this is a bit verbose, but the obvious suggestion of renaming `client` -> `azure` (with `azure.ClientSetConfig`) might not be that good idea either because I see this already in use from `go-autorest` library. I cannot figure better so I guess this is fine.